source/tests/unit/xnet_converters_tests.cpp

```diff
@@ -1,830 +1,116 @@
-  auto data_ptr = converter.data();
-  *data_ptr = IP_INFO_BUFFER.data();
-
-  EXPECT_CALL(library, IpStackFreeAllStacksInfoStr(IP_INFO_BUFFER.data()));
-  auto converted_data = std::string{};
-  converter.to_grpc(converted_data);
-
-  EXPECT_EQ(IP_INFO_STRING, converted_data);
-}
-
-TEST(XnetConvertersTests, AddrWithIPv4Address_ConvertFromGrpc_ConvertsToNxInAddr)
-{
-  constexpr auto ADDRESS = 0x66778899;
-  auto addr = nixnetsocket_grpc::Addr{};
-  addr.mutable_ipv4()->set_addr(ADDRESS);
-
-  const auto converted_data = convert_from_grpc<void>(addr);
-  const auto data_ptr = reinterpret_cast<const nxin_addr*>(static_cast<const void*>(converted_data));
-
-  EXPECT_EQ(ADDRESS, data_ptr->addr);
-}
-
-TEST(XnetConvertersTests, AddrWithIPv6Address_ConvertFromGrpc_ConvertsToNxIn6Addr)
-{
-  const auto ADDRESS = std::vector<char>{0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA, 0xB, 0xC, 0xD, 0xE, 0xF};
-  auto addr = nixnetsocket_grpc::Addr{};
-  addr.mutable_ipv6()->set_addr({ADDRESS.data(), ADDRESS.size()});
-
-  const auto converted_data = convert_from_grpc<void>(addr);
-  const auto data_ptr = reinterpret_cast<const nxin6_addr*>(static_cast<const void*>(converted_data));
-
-  EXPECT_THAT(
-      data_ptr->addr,
-      ElementsAreArray(ADDRESS.data(), ADDRESS.size()));
-}
-
-TEST(XnetConvertersTests, AddrWithNoAddress_ConvertFromGrpc_ConvertsToNxIn6Addr)
-{
-  const auto ADDRESS = std::vector<char>{0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xA, 0xB, 0xC, 0xD, 0xE, 0xF};
-  const auto ZERO_ADDRESS = std::vector<char>(ADDRESS.size(), 0x0);
-  auto addr = nixnetsocket_grpc::Addr{};
-
-  const auto converted_data = convert_from_grpc<void>(addr);
-  const auto data_ptr = reinterpret_cast<const nxin6_addr*>(static_cast<const void*>(converted_data));
-
-  EXPECT_THAT(
-      data_ptr->addr,
-      ElementsAreArray(ZERO_ADDRESS.data(), ZERO_ADDRESS.size()));
-}
-
-TEST(XnetConvertersTests, GrpcAddrCase_GetAddressFamily_ReturnsCorrectDriverFamily)
-{
-  using AddrCase = nixnetsocket_grpc::Addr::AddrCase;
-  EXPECT_EQ(nxAF_UNSPEC, get_address_family(AddrCase::ADDR_NOT_SET));
-  EXPECT_EQ(nxAF_INET, get_address_family(AddrCase::kIpv4));
-  EXPECT_EQ(nxAF_INET6, get_address_family(AddrCase::kIpv6));
-}
-
-TEST(XnetConvertersTests, GrpcIpv4Addr_ConvertFromGrpc_ConvertsToCorrectNxInAddr)
-{
-  constexpr auto ADDRESS = 0x55443322;
-  auto addr = nixnetsocket_grpc::InAddr{};
-  addr.set_addr(ADDRESS);
-
-  const auto converted_data = convert_from_grpc<nxin_addr>(addr);
-
-  EXPECT_EQ(ADDRESS, converted_data.addr);
+TEST(XnetConvertersTests, CANCommStateValue_SetCANCommResponse_ExtractBitfields)
+{
+  u32 canCommunicationState = 0;
+  u32 transcieverError = 0;
+  u32 sleep = 0;
+  u32 lastError = 3;
+  u32 transmitErrorCounter = 3;
+  u32 receiveErrorCounter = 3;
+  // This value is obtained by combining above bit fields.
+  u32 canCommStateValue = 50529024;
```

Comment:
Note: the relationship is a lot clearer in hex (`0x03030300`). You'd still need the comment, but it would be easier to believe.

Reply:
Yeah renamed the state values as hex values.